src/api/storage.py

```
@@ -0,0 +1,41 @@
+
+    headers = {
+        "Authorization": f"Bearer {supabase_service_key}",
+        "apikey": supabase_service_key,
+    }
+    if file.content_type:
+        headers["Content-Type"] = file.content_type
+
+    try:
+        resp = requests.put(upload_url, data=file.file, headers=headers, timeout=10)
+    except requests.RequestException as e:
+        raise HTTPException(500, detail=f"Upload failed: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)